integration_test/third_party_apps_data/simulacra.go

```diff
@@ -0,0 +1,140 @@
+	"strings"
+
+	"github.com/GoogleCloudPlatform/ops-agent/apps"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	"github.com/GoogleCloudPlatform/ops-agent/integration_test/agents"
+	"github.com/GoogleCloudPlatform/ops-agent/integration_test/logging"
+	"github.com/google/uuid"
+
+	"github.com/GoogleCloudPlatform/ops-agent/integration_test/gce"
+)
+
+//go:embed applications
+var scriptsDir embed.FS
+
+func distroFolder(platform string) (string, error) {
+	if gce.IsWindows(platform) {
+		return "windows", nil
+	}
+	firstWord := strings.Split(platform, "-")[0]
+	switch firstWord {
+	case "centos", "rhel", "rocky":
+		return "centos_rhel", nil
+	case "debian", "ubuntu":
+		return "debian_ubuntu", nil
+	case "opensuse", "sles":
+		return "sles", nil
+	}
+	return "", fmt.Errorf("distroFolder() could not find matching folder holding scripts for platform %s", platform)
+}
+
+func setupOpsAgent(ctx context.Context, vm *gce.VM, logger *logging.DirectoryLogger, configFilePath string) error {
+	data, err := ioutil.ReadFile(configFilePath)
+	if err != nil {
+		return err
+	}
+
+	config, err := confgenerator.UnmarshalYamlToUnifiedConfig(ctx, data)
+	if err != nil {
+		return err
+	}
+
+	if err := agents.SetupOpsAgent(ctx, logger.ToMainLog(), vm, config.String()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getAllReceivers(config *confgenerator.UnifiedConfig) (receivers []string) {
+	for logs := range config.Logging.Receivers {
+		app := config.Logging.Receivers[logs].Type()
+		receivers = append(receivers, app)
+	}
+
+	for logs := range config.Metrics.Receivers {
+		app := config.Metrics.Receivers[logs].Type()
+		receivers = append(receivers, app)
+	}
+	return receivers
+}
+
+// installApps reads an Ops Agent config file and then identifies all the third party apps that need to be installed.
+// The function identifies third party apps to install by checking if any of the receiver types have a
+// corresponding install script in the third_party_apps_data directory.
+// If there is a corresponding install script, then that install script is run on the vm.
+func installApps(ctx context.Context, vm *gce.VM, logger *logging.DirectoryLogger, configFilePath string) error {
+	config, err := confgenerator.MergeConfFiles(ctx, configFilePath, apps.BuiltInConfStructs)
+	if err != nil {
+		return err
+	}
+
+	folder, err := distroFolder(vm.Platform)
+
+	if err != nil {
+		return err
+	}
+
+	receivers := getAllReceivers(config)
+
+	for _, app := range receivers {
+		if scriptContent, err := scriptsDir.ReadFile(path.Join("applications", app, folder, "install")); err == nil {
+			logger.ToMainLog().Printf("Installing %s to VM", app)
+			if _, err := gce.RunScriptRemotely(ctx, logger, vm, string(scriptContent), nil, make(map[string]string)); err != nil {
+				return err
+			} else {
+				logger.ToMainLog().Printf("Done Installing %s", app)
+			}
+		}
+	}
+	return nil
+}
+
+func main() {
+	logger, err := logging.NewDirectoryLogger(path.Join("logs"))
```

Comment:
I think it would be better for this tool just to log to stdout rather than using the directory logger. If people need the output of this tool to go to a file they have the option to redirect it in the shell.

Reply:
That's what i had initially, but changed it to be able to call gce.RunScriptRemotely which is how the integration tests execute the install scripts. 

The RunScriptRemotely function is expecting a directory logger. After inspecting the code, I think we can just instead call gce.RunRemotely which takes in a standard logger

Reply:
Ended up doing that, there is some code duplication between simulacra.go and gce_testing.go. We can get rid of the code duplication by fixing the dependency on Directory Logger in RunScriptRemotely in gce_testing.go.